saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/serviceprovider/provider/RelyingPartyRegistration.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.saml2.serviceprovider.provider;
+
+import org.springframework.security.saml2.credentials.Saml2X509Credential;
+import org.springframework.security.saml2.credentials.Saml2X509Credential.Saml2X509CredentialUsage;
+
+import java.net.URI;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Set;
+
+import static java.util.Arrays.asList;
+import static org.springframework.util.Assert.hasText;
+import static org.springframework.util.Assert.notEmpty;
+import static org.springframework.util.Assert.notNull;
+
+/**
+ * Represents a configured service provider, SP, and a remote identity provider, IDP, pair.
+ * Each SP/IDP pair is uniquely identified using a <code>registrationId</code>, and arbitrary string.
+ * A fully configured registration may look like
+ * <pre>
+ *		//remote IDP entity ID
+ *		String idpEntityId = "https://simplesaml-for-spring-saml.cfapps.io/saml2/idp/metadata.php";
+ *		//remote WebSSO Endpoint - Where to Send AuthNRequests to
+ *		String webSsoEndpoint = "https://simplesaml-for-spring-saml.cfapps.io/saml2/idp/SSOService.php";
+ *		//local registration ID
+ *		String registrationId = "simplesamlphp";
+ *		//local entity ID - autogenerated based on URL
+ *		String localEntityIdTemplate = "{baseUrl}/saml2/service-provider-metadata/{registrationId}";
+ *		//local signing (and local decryption key and remote encryption certificate)
+ *		Saml2X509Credential signingCredential = getSigningCredential();
+ *		//IDP certificate for verification of incoming messages
+ *		Saml2X509Credential idpVerificationCertificate = getVerificationCertificate();
+ *		RelyingPartyRegistration rp = new RelyingPartyRegistration(
+ *				idpEntityId,
+ *				registrationId,
+ *				new URI(webSsoEndpoint),
+ *				Arrays.asList(signingCredential, idpVerificationCertificate),
+ *				localEntityIdTemplate
+ *		);
+ * </pre>
+ * @since 5.2
+ */
+public class RelyingPartyRegistration {
```

Comment:
This has a lot of required properties. Please add a Builder and remove the public constructor

Reply:
Builder added